layers/+completion/auto-completion/packages.el

```diff
@@ -128,6 +129,11 @@
                                             "company-statistics-cache.el"))
       (add-hook 'company-mode-hook 'company-statistics-mode))))
 
+(defun auto-completion/post-init-counsel ()
```

Comment:
Use a `pre-init` function instead to setup use-package hooks, using a `post-init` does not guarantee the hook will be setup before the `:config` part of the targeted  package(in this case company) is executed (for instance the package is not lazy loaded).

Said differently, all `pre-init` functions are executed before the `init` functions so we are 100% guaranteed that the hooks are installed before the `init` function is executed.

Reply:
You could update `auto-completion/init-helm-company`with this better idiom as well :-)

Reply:
Oh to update `auto-completion/init-helm-company` you'll be able to create a `pre-init` function as well as an empty `init` function. `init` function is mandatory, if there is no `init` function for a package then it is considered unused. Also the layer defining the `init` function of a package is called the owner of this package.

This rule with `init` function may be simplified at some point :-)

Reply:
You can see an example of this here: https://github.com/syl20bnr/spacemacs/blob/develop/layers/%2Blang/elixir/packages.el#L172

Reply:
Ok, i pushed your requested changes.